Firmware/ultralcd.cpp

```diff
@@ -5191,7 +5200,9 @@ static void lcd_main_menu()
     } else if (!Stopped) {
         MENU_ITEM_SUBMENU_P(_i("Preheat"), lcd_preheat_menu);////MSG_PREHEAT c=18
     }
-
+    if (printer_state == 0) {
```

Comment:
I feel this menu item should be hidden if there is no host connected/enabled. Can we check that somehow?

If user has no remote host and uses only SD card this menu won’t do anything. In that case I think we should try minimize amount of menu items to simplify the user interface.

Reply:
Right, there will be a new USB PrusaLink connected state to be checked. It have been better to start with that.
BTW the `Set Ready`on MK4 is atm also shown even without LAN/WiFi connected.